spec/services/void_participant_declaration_service_spec.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+require_relative "../shared/context/service_record_declaration_params.rb"
+require_relative "../shared/context/lead_provider_profiles_and_courses.rb"
+
+RSpec.describe VoidParticipantDeclaration do
+  include_context "lead provider profiles and courses"
+  include_context "service record declaration params"
+
+  context "when voiding declaration" do
+    let(:start_date) { ect_profile.schedule.milestones.first.start_date }
+    before do
+      travel_to start_date + 2.days
+      RecordParticipantDeclaration.call(ect_params)
+      travel_to start_date + 3.days
+    end
+
+    it "voids a participant declaration" do
+      declaration = ParticipantDeclaration.order(:declaration_date).first
+      described_class.new(cpd_lead_provider: cpd_lead_provider, id: declaration.id).call
+      expect(declaration.reload.voided).to be_truthy
+    end
+
+    it "does not void a voided declaration" do
+      declaration = ParticipantDeclaration.order(:created_at).last
+      described_class.new(cpd_lead_provider: cpd_lead_provider, id: declaration.id).call
+      expect {
+        described_class.new(cpd_lead_provider: cpd_lead_provider, id: declaration.id).call
+      }.to raise_error ActionController::BadRequest
```

Comment:
Why BadRequest? Surely it's an Unprocessable entity. The form of the request is ok, it's just the content.